addons/l10n_nz/migrations/1.2/pre-migrate.py

```diff
@@ -0,0 +1,11 @@
+def migrate(cr, version):
+    cr.execute("""
+        UPDATE account_report_expression
+        SET label = 'temporary_label'
+        WHERE id = (
+            SELECT res_id
+            FROM ir_model_data
+            WHERE module = 'l10n_nz'
+            AND name = 'tax_report_box13_formula'
+        );
```

Comment:
```suggestion
        UPDATE account_report_expression e
           SET label = '_upg_1.2_label'
          FROM ir_model_data d
         WHERE e.id = d.res_id
           AND d.module = 'l10n_nz'
           AND d.name = 'tax_report_box13_formula'
           AND e.label = 'balance'
```
Then in `post-` check if the prefix `_upg_1.2_` is still there, then revert it. It could happen if the record was marked as noupdate.

Reply:
We can also force the `noupdate` flag of the record

Reply:
Why would we do that ? I'll test again, but from what I saw during my previous tests, the labels were updated automatically (I couldn't find any "temporary_label" in the DB), so we could do without a post-migrate

Reply:
You are not upgrading just standard data. This will be client's data. They could have updated this in ways that could fail.

Reply:
Also this shouldn't be necessary for master. As long as the forward ports from https://github.com/odoo/odoo/pull/194812 are merged. (cc: @KangOl)

Reply:
> Also this shouldn't be necessary for master

Seems like they are needed in the end (upgrade 18.0->master failing because of the local upgrade failing)